gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -33,90 +33,100 @@ var fluid = fluid || require("infusion");
             }
         },
         invokers: {
-            grantAuthorizationCode: {
-                funcName: "gpii.oauth2.authorizationService.grantAuthorizationCode",
+            // APIs for GPII App Installation Clients
+            grantGpiiAppInstallationAuthorization: {
+                funcName: "gpii.oauth2.authorizationService.grantGpiiAppInstallationAuthorization",
+                args: ["{dataStore}", "{codeGenerator}", "{arguments}.0", "{arguments}.1"]
+                    // gpiiToken, clientId
+            },
+
+            // APIs for Web Preferences Consumer Clients
+            exchangeCodeForAccessToken: {
+                funcName: "gpii.oauth2.authorizationService.exchangeCodeForAccessToken",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // code, clientId, redirectUri
+            },
+            grantWebPrefsConsumerAuthCode: {
+                funcName: "gpii.oauth2.authorizationService.grantWebPrefsConsumerAuthCode",
                 args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2", "{arguments}.3"]
                 // args: ["{dataStore}", "{codeGenerator}", "{arguments}.0", "{arguments}.1", "{arguments}.2", "{arguments}.3"]
                     // gpiiToken, clientId, redirectUri, selectedPreferences
             },
-            addAuthorization: {
-                funcName: "gpii.oauth2.authorizationService.addAuthorization",
-                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
-                    // gpiiToken, clientId, selectedPreferences
-            },
-            userHasAuthorized: {
-                funcName: "gpii.oauth2.authorizationService.userHasAuthorized",
+            userHasAuthorizedWebPrefsConsumer: {
+                funcName: "gpii.oauth2.authorizationService.userHasAuthorizedWebPrefsConsumer",
                 args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
                     // gpiiToken, clientId, redirectUri
             },
-            exchangeCodeForAccessToken: {
-                funcName: "gpii.oauth2.authorizationService.exchangeCodeForAccessToken",
-                args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
-                    // code, clientId, redirectUri
+
+            // APIs for both Web Preferences Consumer Clients and Onboarded Solution Clients
+            addUserAuthorizedAuthorization: {
+                funcName: "gpii.oauth2.authorizationService.addUserAuthorizedAuthorization",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2", "{arguments}.3"]
```

Comment:
This doesn't match the actual function definition which only accepts 3 arguments (+ that)